browser/ui/webui/brave_wallet/trezor_bridge_ui.cc

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/brave_wallet/trezor_bridge_ui.h"
+
+#include <memory>
+#include <utility>
+
+#include "brave/browser/ui/webui/brave_webui_source.h"
+#include "brave/common/webui_url_constants.h"
+#include "brave/components/trezor_bridge/resources/grit/trezor_bridge_generated_map.h"
+#include "brave/components/trezor_bridge/trezor_bridge_handler.h"
+#include "components/grit/brave_components_resources.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/storage_partition.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "content/public/browser/web_ui_message_handler.h"
+#include "ui/base/resource/resource_bundle.h"
+#include "ui/resources/grit/webui_generated_resources.h"
+
+using content::WebUIMessageHandler;
+
+TrezorBridgeUI::TrezorBridgeUI(content::WebUI* web_ui, const std::string& name)
+    : WebUIController(web_ui) {
+  auto* html_source = CreateAndAddWebUIDataSource(
+      web_ui, name, kTrezorBridgeGenerated, kTrezorBridgeGeneratedSize,
+      IDR_TREZOR_BRIDGE_HTML);
+  html_source->AddResourcePath("trezor/iframe.html", IDR_TREZOR_BRIDGE_IFRAME);
+
+  html_source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::FrameSrc,
+      "frame-src chrome://trezor-bridge;");
+  html_source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ScriptSrc,
+      "script-src chrome://resources/ chrome://trezor-bridge ;");
+  // Disable DenyXFrame to allow TrezorConnect to create iframes.
+  html_source->DisableDenyXFrameOptions();
```

Comment:
i suggest setting the `frame-ancestors` CSP directive so it can only be framed by whitelisted origins

Reply:
done, thanks!

Reply:
I think @diracdeltas meant (1) add `frame-ancestors` and (2) remove the `html_source->DisableDenyXFrameOptions();`. They should be mutually exclusive.